internal/mongodoc/hash.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package mongodoc
+
+import (
+	"crypto/sha512"
+	"encoding/hex"
+
+	"gopkg.in/errgo.v1"
+	"gopkg.in/mgo.v2/bson"
+)
+
+// Hashes holds a slice of hash objects
+// that will encode to MongoDB in a space-efficient
+// way. Each of the strings must be a hex-encoded
+// SHA384 checksum as produced by blobstore.NewHash.
```

Comment:
I'm still very dubious about the value of this, whilst the space saving may be worth it it will make it significantly harder to understand the mongodb document in isolation. I'm not certain that is a price worth paying.